fulp_modules/bloodsuckers/bloodsucker/bloodsucker.dm

```diff
@@ -0,0 +1,746 @@
+
+/datum/antagonist/bloodsucker/proc/LevelUpPowers()
+	for(var/datum/action/bloodsucker/power in powers)
+		power.level_current ++
+
+/datum/antagonist/bloodsucker/proc/SpendRank()
+	set waitfor = FALSE
+	if(bloodsucker_level_unspent <= 0 || !owner || !owner.current || !owner.current.client || !isliving(owner.current))
+		return
+	var/mob/living/L = owner.current
+	level_bloodcost = max_blood_volume * 0.2
+	//If the blood volume of the bloodsucker is lower than the cost to level up, return and inform the bloodsucker
+	//TODO: Make this into a radial, or perhaps a tgui next UI
+		// Purchase Power Prompt
+	var/list/options = list()
+	for(var/pickedpower in typesof(/datum/action/bloodsucker))
+		var/datum/action/bloodsucker/power = pickedpower
+		// If I don't own it, and I'm allowed to buy it.
+		if(!(locate(power) in powers) && initial(power.bloodsucker_can_buy))
+			options[initial(power.name)] = power // TESTING: After working with TGUI, it seems you can use initial() to view the variables inside a path?
+	options["\[ Not Now \]"] = null
+	// Abort?
+	if(options.len > 1)
+		var/choice = input(owner.current, "You have the opportunity to grow more ancient at the cost of [level_bloodcost] units of blood. Select a power to advance your Rank.", "Your Blood Thickens...") in options
+		// Cheat-Safety: Can't keep opening/closing coffin to spam levels
+		if(bloodsucker_level_unspent <= 0) // Already spent all your points, and tried opening/closing your coffin, pal.
+			return
+		if(!istype(owner.current.loc, /obj/structure/closet/crate/coffin))
+			to_chat(owner.current, "<span class='warning'>Return to your coffin to advance your Rank.</span>")
+			return
+		if(!choice || !options[choice] || (locate(options[choice]) in powers)) // ADDED: Check to see if you already have this power, due to window stacking.
+			to_chat(owner.current, "<span class='notice'>You prevent your blood from thickening just yet, but you may try again later.</span>")
+			return
+		if(L.blood_volume < level_bloodcost)
+			to_chat(owner.current, "<span class='warning'>You dont have enough blood to thicken your blood, you need [level_bloodcost - L.blood_volume] units more!</span>")
+			return
+		// Buy New Powers
+		var/datum/action/bloodsucker/P = options[choice]
+		AddBloodVolume(-level_bloodcost)
+		BuyPower(new P)
+		to_chat(owner.current, "<span class='notice'>You have used [level_bloodcost] units of blood and learned [initial(P.name)]!</span>")
+	else
+		to_chat(owner.current, "<span class='notice'>You grow more ancient by the night!</span>")
+	/////////
+	// Advance Powers (including new)
+	LevelUpPowers()
+	////////
+	// Advance Stats
+	if(ishuman(owner.current))
+		var/mob/living/carbon/human/H = owner.current
+		var/datum/species/S = H.dna.species
+		S.punchdamagelow += 0.5
+		S.punchdamagehigh += 0.5    // NOTE: This affects the hitting power of Brawn.
```

Comment:
seems kinda underwhelming

Reply:
Yeah, it's meant to get stronger overtime.